go/controller/internal/common/bindings.go

```diff
@@ -0,0 +1,57 @@
+package common
+
+import (
+	console "github.com/pluralsh/console/go/client"
+	"github.com/pluralsh/console/go/controller/api/v1alpha1"
+	"github.com/pluralsh/console/go/controller/internal/identity"
+	"github.com/samber/lo"
+)
+
+// BindingsAttributes translates a list of bindings into API attributes using identity cache.
+func BindingsAttributes(bindings []v1alpha1.Binding) ([]*console.PolicyBindingAttributes, error) {
+	if bindings == nil {
+		return nil, nil
+	}
+
+	attrs := make([]*console.PolicyBindingAttributes, 0)
+	for _, b := range bindings {
+		attr, err := bindingAttributes(b)
+		if err != nil {
+			return nil, err
+		}
+		if attr != nil {
+			attrs = append(attrs, attr)
+		}
+	}
+
+	return attrs, nil
+}
+
+// bindingAttributes translates a binding into API attributes using identity cache.
+// This couldn't be done directly on the binding type to avoid circular imports.
+func bindingAttributes(b v1alpha1.Binding) (*console.PolicyBindingAttributes, error) {
+	userId := b.UserID
+	groupId := b.GroupID
+
+	if userId == nil && b.UserEmail != nil {
+		id, err := identity.Cache().GetUserID(*b.UserEmail)
+		if err != nil {
+			return nil, err
+		}
+		userId = lo.EmptyableToPtr(id)
+	}
+
+	if groupId == nil && b.GroupName != nil {
+		id, err := identity.Cache().GetGroupID(*b.GroupName)
+		if err != nil {
+			return nil, err
+		}
+		groupId = lo.EmptyableToPtr(id)
+	}
+
+	if userId == nil && groupId == nil {
+		return nil, nil
+	}
+
+	return &console.PolicyBindingAttributes{ID: b.ID, UserID: userId, GroupID: groupId}, nil
```

Comment:
I don't think you should set the `ID` here, unless you know its the id that's coming from our api.  If it's not a uuid, I'm pretty sure we'll throw for instance.

Reply:
This was never coming from our API. It was just there in the CRD as far as I am aware. I think it should be safe to remove it from here an deprecate it in the CRD spec. Does that make sense?